src/plasma/eviction_policy.cc

```diff
@@ -1,109 +1,75 @@
 #include "eviction_policy.h"
 
 #include "utlist.h"
+#include "uthash.h"
 
-/** An element representing a released object in a doubly-linked list. This is
- *  used to implement an LRU cache. */
-typedef struct ReleasedObject {
-  /** The object_id of the released object. */
-  ObjectID object_id;
-  /** Needed for the doubly-linked list macros. */
-  struct ReleasedObject *prev;
-  /** Needed for the doubly-linked list macros. */
-  struct ReleasedObject *next;
-} ReleasedObject;
-
-/** This type is used to define a hash table mapping the object ID of a released
- *  object to its location in the doubly-linked list of released objects. */
-typedef struct {
-  /** Object ID of this object. */
-  ObjectID object_id;
-  /** A pointer to the corresponding entry for this object in the doubly-linked
-   *  list of released objects. */
-  ReleasedObject *released_object;
-  /** Handle for the uthash table. */
-  UT_hash_handle handle;
-} released_object_entry;
+#include <list>
+#include <unordered_map>
+
+class LRUCache {
+ private:
+  // A doubly-linked list containing the items in the cache and
+  // their sizes in LRU order.
+  std::list<std::pair<ObjectID, int64_t>> item_list_;
+  // A hash table mapping the object ID of objects in the cache to its location
+  // in the doubly linked list item_list_.
+  std::unordered_map<ObjectID, decltype(item_list_.begin()), UniqueIDHasher>
+      item_map_;
+
+ public:
+  LRUCache(){};
+
+  void store(const ObjectID &key, int64_t size) {
+    auto it = item_map_.find(key);
+    if (it == item_map_.end()) {
+      item_list_.push_front(std::make_pair(key, size));
+      item_map_.insert(std::make_pair(key, item_list_.begin()));
+    }
+  }
+
+  void remove(const ObjectID &key) {
+    auto it = item_map_.find(key);
+    if (it != item_map_.end()) {
+      item_list_.erase(it->second);
+      item_map_.erase(it);
+    }
+  }
+
+  void access(const ObjectID &key) {
```

Comment:
It brings the object to the front of the cache. I want to do some refactoring in the future that uses this, for now I removed it.